writeToStream.js

```diff
@@ -200,7 +200,7 @@ function connect (packet, stream, opts) {
 
   // Password
   if (password != null) {
-    if (!providedUsername) {
+    if (!providedUsername && protocolVersion !== 5) {
```

Comment:
Could you add a comment with a link to official docs that explain why this?

Reply:
I'm new in javascript.
Plz wait for test cases.